main/src/vtr_radar/src/modules/preprocessing/conversions/RAS3_extraction_module.cpp

```diff
@@ -0,0 +1,350 @@
+  config->modified_cacfar.guard = node->declare_parameter<int>(param_prefix + ".modified_cacfar.guard", config->modified_cacfar.guard);
+  config->modified_cacfar.threshold = node->declare_parameter<double>(param_prefix + ".modified_cacfar.threshold", config->modified_cacfar.threshold);
+  config->modified_cacfar.threshold2 = node->declare_parameter<double>(param_prefix + ".modified_cacfar.threshold2", config->modified_cacfar.threshold2);
+  config->modified_cacfar.threshold3 = node->declare_parameter<double>(param_prefix + ".modified_cacfar.threshold3", config->modified_cacfar.threshold3);
+
+  // added new cell averging smallest of detector
+  config->caso_cfar.width = node->declare_parameter<int>(param_prefix + ".caso_cfar.width", config->caso_cfar.width);
+  config->caso_cfar.guard = node->declare_parameter<int>(param_prefix + ".caso_cfar.guard", config->caso_cfar.guard);
+  config->caso_cfar.threshold = node->declare_parameter<double>(param_prefix + ".caso_cfar.threshold", config->caso_cfar.threshold);
+  
+
+
+  config->radar_resolution = node->declare_parameter<double>(param_prefix + ".radar_resolution", config->radar_resolution);
+  config->cart_resolution = node->declare_parameter<double>(param_prefix + ".cart_resolution", config->cart_resolution);
+
+  // // Doppler stuff
+  // config->beta = node->declare_parameter<double>(param_prefix + ".beta", config->beta);
+  // config->chirp_type = node->declare_parameter<std::string>(param_prefix + ".chirp_type", config->chirp_type);
+
+  config->visualize = node->declare_parameter<bool>(param_prefix + ".visualize", config->visualize);
+  // clang-format on
+  return config;
+}
+
+void RAS3ExtractionModule::run_(QueryCache &qdata0, OutputCache &,
+                                   const Graph::Ptr &,
+                                   const TaskExecutor::Ptr &) {
+  auto &qdata = dynamic_cast<RadarQueryCache &>(qdata0);
+
+  /// Create a node for visualization if necessary
+  if (config_->visualize && !publisher_initialized_) {
+    // clang-format off
+    scan_pub_ = qdata.node->create_publisher<ImageMsg>("raw_scan", 5);
+    fft_scan_pub_ = qdata.node->create_publisher<ImageMsg>("fft_scan", 5);
+    bev_scan_pub_ = qdata.node->create_publisher<ImageMsg>("bev_scan", 5);
+    pointcloud_pub_ = qdata.node->create_publisher<PointCloudMsg>("raw_point_cloud", 5);
+    // clang-format on
+    publisher_initialized_ = true;
+  }
+
+  if(!qdata.scan_msg)
+  {
+    return;
+  }
+
+  /// Input radar message from the query cache
+  // This is to convert a ROS image to an opencv image
+  // CLOG(DEBUG, "radar.navtech_extractor") << "Sam: " << qdata.scan_msg->b_scan_img.encoding;
+  auto fft_scan = cv_bridge::toCvCopy(std::make_shared<ImageMsg>(qdata.scan_msg->b_scan_img))->image;
+
+  fft_scan.convertTo(fft_scan, CV_32F); 
+
+  // normalize to 0-1
+  fft_scan = fft_scan/255.0;
+
+  /// Output
+  auto &beta = *qdata.beta.emplace();
```

Comment:
Remove all references to doppler in this file.

Reply:
beta is used so I will comment out for future uses